File: modules/model/src/main/java/no/ntnu/remotecode/model/Template.java
package no.ntnu.remotecode.model;


import lombok.Data;
import lombok.NoArgsConstructor;

import javax.json.bind.annotation.JsonbTransient;
import javax.persistence.*;
import java.util.List;
import java.util.UUID;

/**
 * Represents an template for creating new containers.
 */

@Entity
@Data
@NoArgsConstructor
@Inheritance(strategy = InheritanceType.JOINED)
@Table(name = "container_template")
public class Template {


    @JsonbTransient
    @javax.persistence.Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long Id;

    /**
     * The uuid key for this container.
     */
    @Column(unique = true)
    private UUID templateKey;


    /**
     * The id of the admin user who created the template.
     */
    @JsonbTransient
    private double creatorId;


    /**
     * The name of the template used.
     */
    private String templateName;

    /**
     * The name of the template dir containing the build docker file
     */
    @JsonbTransient
    private String buildDirName;

    /**
     * The name of the built image used for this template.
     */
    @JsonbTransient
    private String TemplateImageName;


    /**
     * The repo to clone in to the projects built from this template
     */
    private String gitCloneRepo;

}

File: modules/frontend/vue-frontend/src/models/Template.ts
export class Template {
  private _templateKey: string;
  private _templateName: string;
  private _gitCloneRepo: string;
  private _templateLink: string;

  constructor (templateKey: string, templateName: string, gitCloneRepo: string, templateLink: string) {
    this._templateKey = templateKey
    this._templateName = templateName
    this._gitCloneRepo = gitCloneRepo
    this._templateLink = templateLink
  }

  get templateKey (): string {
    return this._templateKey
  }

  get templateName (): string {
    return this._templateName
  }

  get gitCloneRepo (): string {
    return this._gitCloneRepo
  }

  get templateLink (): string {
    return this._templateLink
  }
}
